core/index/index_reader.hpp

```diff
@@ -129,6 +129,8 @@ struct SubReader : public IndexReader, public ColumnProvider {
 
   size_t size() const noexcept final { return 1; }
 
+  virtual void CountMemory(MemoryStats stats) const = 0;
```

Comment:
Why not move it to "IndexReader" ?

Reply:
And would be better to pass value by reference. Copying 3 pointers now is not a big problem. But this stats might grow in future.

Reply:
I think now is better by value.
If size of struct will growth a lot, we could change this

Reply:
Ok